services/traction/api/db/models/base.py

```diff
@@ -36,3 +81,28 @@ class BaseTable(BaseModel):
             TIMESTAMP, nullable=False, server_default=func.now(), onupdate=func.now()
         )
     )
+
+
+class TimestampModel(BaseModel):
+    created_at: datetime = Field(
+        sa_column=Column(TIMESTAMP, nullable=False, server_default=func.now())
+    )
+    updated_at: datetime = Field(
+        sa_column=Column(
+            TIMESTAMP, nullable=False, server_default=func.now(), onupdate=func.now()
+        )
+    )
+
+
+class StatefulModel(BaseModel):
+    status: str = Field(nullable=False)
+    state: str = Field(nullable=False)
+    error_status_detail: str = Field(nullable=True)
```

Comment:
https://docs.sqlalchemy.org/en/14/core/defaults.html

What about this. 

```
error_status_detail:: Field(
        sa_column=Column(
            str, nullable=False, onupdate=_secret_method
        )
    )
    
def _secret_method(context):
    if context.get_current_parameters()["state"] != "error":
       return None
    
    ```

Reply:
if that works, let's put it in. and add it to timeline.  I'll take a look and get it in place.

Reply:
couple of tweaks but works great, removed the trigger that was doing this work
https://github.com/bcgov/traction/blob/4efd739c443f878699774ef79f8633ad1fd75b32/services/traction/api/db/models/base.py#L100